x-pack/metricbeat/mbreceiver/factory.go

```diff
@@ -50,7 +52,20 @@ func createReceiver(_ context.Context, set receiver.Settings, baseCfg component.
 		return nil, fmt.Errorf("error getting %s creator:%w", Name, err)
 	}
 
-	return &metricbeatReceiver{beat: &b.Beat, beater: mbBeater, logger: set.Logger}, nil
+	httpConf := struct {
+		HTTP *config.C `config:"http"`
+	}{}
+	if err := b.RawConfig.Unpack(&httpConf); err != nil {
+		return nil, fmt.Errorf("error starting API: %w", err)
```

Comment:
This is a parsing error, right? The message should reflect that.

Reply:
Thanks.  I fixed it in fbreceiver as well since it is the same there.